packages/flame/test/components/position_component_test.dart

```diff
@@ -285,4 +295,301 @@ void main() {
+      expect(component.positionOf(Vector2(10, 0)), Vector2(44, 22));
+    });
+
+    test('rotations', () {
+      final component = PositionComponent()
+        ..size = Vector2(8, 6)
+        ..position = Vector2(50, 20)
+        ..anchor = Anchor.center;
+
+      // Rotate the component in small increments counterclockwise
+      // and track the coordinate of its top-right corner
+      for (var i = 0; i < 30; i++) {
+        component.angle = -i / 10;
+        final cosA = math.cos(i / 10);
+        final sinA = math.sin(i / 10);
+        final expectedX = 50 + 5 * (0.8 * cosA - 0.6 * sinA);
+        final expectedY = 20 - 5 * (0.6 * cosA + 0.8 * sinA);
+        final topRight = component.positionOf(Vector2(8, 0));
+        expect(topRight.x, closeTo(expectedX, 1e-10));
+        expect(topRight.y, closeTo(expectedY, 1e-10));
+      }
+    });
+
+    test('random local<->global', () {
+      final parent = PositionComponent()..size = Vector2(50, 25);
+      final child = PositionComponent()
+        ..size = Vector2(10, 8)
+        ..position = Vector2(50, 20)
+        ..anchor = const Anchor(0.1, 0.2);
+      parent.addChild(child);
+
+      final rnd = math.Random();
+      for (var i = 0; i < 100; i++) {
+        child.angle = (rnd.nextDouble() - 0.5) * 10;
+        child.x = rnd.nextDouble() * 100;
+        child.y = rnd.nextDouble() * 70 + 30;
+        parent.angle = (rnd.nextDouble() - 0.5) * 5;
+        parent.x = rnd.nextDouble() * 300;
+        parent.y = rnd.nextDouble() * 1000;
+        if (rnd.nextBool()) {
+          child.flipHorizontally();
+        }
+        if (rnd.nextBool()) {
+          child.flipVertically();
+        }
+        if (rnd.nextDouble() < 0.1) {
+          parent.flipHorizontally();
+        }
+
+        final globalX = (rnd.nextDouble() - 0.3) * 200;
+        final globalY = (rnd.nextDouble() - 0.1) * 200;
+        final localPoint = child.absoluteToLocal(Vector2(globalX, globalY));
+        final globalPoint = child.absolutePositionOf(localPoint);
+        expect(globalPoint.x, closeTo(globalX, 1e-10));
+        expect(globalPoint.y, closeTo(globalY, 1e-10));
+      }
+    });
+
+    test('transform matrix', () {
+      final component = PositionComponent()
+        ..size = Vector2(5, 10)
+        ..anchor = Anchor.center;
+
+      final rnd = math.Random();
+      for (var i = 0; i < 10; i++) {
+        final x = rnd.nextDouble() * 100;
+        final y = rnd.nextDouble() * 70 + 30;
+        final angle = (rnd.nextDouble() - 0.5) * 10;
+        component.x = x;
+        component.y = y;
+        component.angle = angle;
+
+        final transform = Matrix4.identity()
+          ..translate(x, y)
+          ..rotateZ(angle)
+          ..translate(
+            -component.anchor.x * component.width,
+            -component.anchor.y * component.height,
+          );
+        for (var j = 0; j < 16; j++) {
+          expect(component.transformMatrix[j], closeTo(transform[j], 1e-10));
```

Comment:
I think we have an `expectDouble` util that has a default tolerance

Reply:
I'm sorry, but I don't like that util very much :(

For once, the default tolerance is too high. These computations should easily hold up to 1e-12 precision if not lower. (The machine precision for doubles is 1e-16 near 1, but given compounding effects from multiple calculations and taking into account scale factors, 1e-10 is more than enough precision).

The second reason is that when the test fails, `closeTo` provides a much better description of what happened. For example, this is the error message I get from `expectDouble()`:
```
  Expected: <true>
    Actual: <false>
```
And here's the same error message with `closeTo()`:
```
  Expected: a numeric value within <1e-13> of <99.64973343376249>
    Actual: <99.6497334337626>
     Which:  differs by <1.1368683772161603e-13>
```